ext/opentelemetry-ext-datadog/src/opentelemetry/ext/datadog/__init__.py

```diff
@@ -0,0 +1,211 @@
+        service: The service to be used for the application or use `DD_SERVICE` environment variable
+    """
+
+    def __init__(self, agent_url=None, service=None):
+        self.agent_url = (
+            agent_url
+            if agent_url
+            else os.environ.get("DD_TRACE_AGENT_URL", DEFAULT_AGENT_URL)
+        )
+        self.service = service if service else os.environ.get("DD_SERVICE")
+        self._agent_writer = None
+
+    @property
+    def agent_writer(self):
+        if self._agent_writer is None:
+            url_parsed = urlparse(self.agent_url)
+            if url_parsed.scheme in ("http", "https"):
+                self._agent_writer = AgentWriter(
+                    hostname=url_parsed.hostname,
+                    port=url_parsed.port,
+                    https=url_parsed.scheme == "https",
+                )
+            elif url_parsed.scheme == "unix":
+                self._agent_writer = AgentWriter(uds_path=url_parsed.path)
+            else:
+                raise ValueError(
+                    "Unknown scheme `%s` for agent URL" % url_parsed.scheme
+                )
+        return self._agent_writer
+
+    def export(self, spans):
+        datadog_spans = self._translate_to_datadog(spans)
+
+        self.agent_writer.write(spans=datadog_spans)
+
+        return SpanExportResult.SUCCESS
+
+    def shutdown(self):
+        if self.agent_writer.started:
+            self.agent_writer.stop()
+            self.agent_writer.join(self.agent_writer.exit_timeout)
+
+    def _translate_to_datadog(self, spans):
+        datadog_spans = []
+
+        for span in spans:
+            trace_id, parent_id, span_id = _get_trace_ids(span)
+
+            # datadog Span is initialized with a reference to the tracer which is
+            # used to record the span when it is finished. We can skip ignore this
+            # because we are not calling the finish method and explictly set the
+            # duration.
+            tracer = None
+
+            datadog_span = DatadogSpan(
+                tracer,
+                span.name,
+                service=self.service,
+                resource=_get_resource(span),
+                span_type=_get_span_type(span),
+                trace_id=trace_id,
+                span_id=span_id,
+                parent_id=parent_id,
+            )
+            datadog_span.start_ns = span.start_time
+            datadog_span.duration_ns = span.end_time - span.start_time
+
+            if span.status.canonical_code is not StatusCanonicalCode.OK:
+                datadog_span.error = 1
+                if span.status.description:
+                    exc_type, exc_val = _get_exc_info(span)
+                    # no mapping for error.stack since traceback not recorded
+                    datadog_span.set_tag("error.msg", exc_val)
+                    datadog_span.set_tag("error.type", exc_type)
+
+            datadog_span.set_tags(span.attributes)
+
```

Comment:
Are events missing?

Reply:
@majorgreys it sounds like these map nicely to metrics?

Reply:
> Are events missing?

Yes. I added a note to clarify.